Remove `PKApplePayLaterAvailability` staging code

aprotyas opened the pull request:
#### 6529c84e61106eac2b3bb837cbcb2448b9a01102
<pre>
Remove `PKApplePayLaterAvailability` staging code
<a href="https://bugs.webkit.org/show_bug.cgi?id=255360">https://bugs.webkit.org/show_bug.cgi?id=255360</a>
rdar://107955442

Reviewed by Aditya Keerthi.

This commit removes the staging code and also removes the merchant
ineligible reason from PKApplePayLaterAvailability, mirroring recent
changes to the upstream PK API.

* LayoutTests/http/tests/paymentrequest/paymentrequest-applePayLaterAvailability.https.html:

Use a different PKApplePayLaterAvailability enumeration value. All of
them serve the same purpose in this test context.

* Source/WebCore/Modules/applepay/ApplePayLaterAvailability.h:
* Source/WebCore/Modules/applepay/ApplePayLaterAvailability.idl:
* Source/WebCore/PAL/pal/spi/cocoa/PassKitSPI.h:
* Source/WebKit/Shared/ApplePay/cocoa/WebPaymentCoordinatorProxyCocoa.mm:
(WebKit::toPKApplePayLaterAvailability):
(WebKit::WebPaymentCoordinatorProxy::platformPaymentRequest):

Canonical link: <a href="https://commits.webkit.org/264096@main">https://commits.webkit.org/264096@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a1800a4bbcfbb2ef1dac06187087415cbae58da6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6675 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6891 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7073 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8264 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/6948 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/6674 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/7981 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/6837 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/9846 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6793 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/6111 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8353 "Built successfully") | 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/6039 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/13868 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/6120 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/8825 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6606 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/5411 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6001 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/5967 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1578 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10173 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6374 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on LayoutTests/http/tests/paymentrequest/paymentrequest-applePayLaterAvailability.https.html:
```diff
@@ -46,7 +46,7 @@
 user_activation_test(async (test) => {
     const method = validPaymentMethod();
-    method.data.applePayLaterAvailability = "unavailableMerchantIneligible";
+    method.data.applePayLaterAvailability = "UnavailableRecurringTransaction";
```
Is this case sensitive? The enum value in the IDL below is `unavailableRecurringTransaction`.

aprotyas replied:
Oop it definitely is. Conflated the WebCore enum with the IDL enum, turns out they have different casing. Fixed in https://github.com/WebKit/WebKit/commit/a1800a4bbcfbb2ef1dac06187087415cbae58da6

aprotyas committed:
Remove `PKApplePayLaterAvailability` staging code
https://bugs.webkit.org/show_bug.cgi?id=255360
rdar://107955442

Reviewed by Aditya Keerthi.

This commit removes the staging code and also removes the merchant
ineligible reason from PKApplePayLaterAvailability, mirroring recent
changes to the upstream PK API.

* LayoutTests/http/tests/paymentrequest/paymentrequest-applePayLaterAvailability.https.html:

Use a different PKApplePayLaterAvailability enumeration value. All of
them serve the same purpose in this test context.

* Source/WebCore/Modules/applepay/ApplePayLaterAvailability.h:
* Source/WebCore/Modules/applepay/ApplePayLaterAvailability.idl:
* Source/WebCore/PAL/pal/spi/cocoa/PassKitSPI.h:
* Source/WebKit/Shared/ApplePay/cocoa/WebPaymentCoordinatorProxyCocoa.mm:
(WebKit::toPKApplePayLaterAvailability):
(WebKit::WebPaymentCoordinatorProxy::platformPaymentRequest):

Canonical link: https://commits.webkit.org/264096@main
